### PR TITLE
PIMS-599 Map Filter

### PR DESCRIPTION
**Pull request by asanchezr**
- add building blocks for frontend forms
- align frontend service calls with backend models
- add icon button component

### CODE REVIEW

File: frontend/src/components/common/form/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+/**
+ * Formik-connected form control that provides thin wrappers for React-Bootstrap form controls;
+ * e.g. <Form.Row>, <Form.Group>, <Form.Control>, etc.
+ */
+const Form = ({ className, children, ...rest }: FormProps) => {
+  const { handleSubmit } = useFormikContext();
+  return (
+    <FormBase noValidate className={className} onSubmit={handleSubmit} {...(rest as any)}>
```
Comment (Fosol) [inline frontend/src/components/common/form/Form.tsx:20]: Can you just go `{...any}`?

File: frontend/src/components/common/form/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
+import React from 'react';
```
Comment (Fosol) [inline frontend/src/components/common/form/Select.tsx:1]: Does this support dynamic list?

File: frontend/src/utils/utils.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+export const classSet = (input?: any): string => {
```
Comment (Fosol) [inline frontend/src/utils/utils.ts:1]: Can you add some documentation `jsdoc` to these?

File: frontend/src/actionCreators/parcelsActionCreator.spec.tsx [hunk 4]
```diff
@@ -39,8 +39,11 @@ describe('fetchParcels action creator', () => {
       neLongitude: 2,
       swLatitude: 3,
       swLongitude: 4,
-      agencyId: null,
-      propertyClassificationId: null,
+      address: null,
```
[the remark this answers is not in the document]
Comment (asanchezr) [reply frontend/src/actionCreators/parcelsActionCreator.spec.tsx:42]: as far as I know `address` is a string and that's what the frontend sends. Strings can be `null` in which case the API call looks like: `/api/parcels?address=&otherparams...`

File: frontend/src/components/common/form/DisplayError.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (asanchezr) [reply frontend/src/components/common/form/DisplayError.tsx:1]: yes. the idea is that you describe the "shape" of your form via `Yup` (an schema library) and `<Formik>` (the form library chosen for the frontend) will attach error messages based on the form schema defined. We can go into more details on our next dev training session.

File: frontend/src/components/common/form/Form.tsx (hunk 1 above)
Comment (asanchezr) [reply frontend/src/components/common/form/Form.tsx:20]: Nope. The `as any` is to "turn off" the types inferred by typescript. This is a common pattern in React where creating a custom typing when you are "extending" a base class (like here we are wrapping the react-bootstrap <Form>). It's fine as in JavaScript the types are removed and it ends up being `{...rest}` which is what we want

File: frontend/src/utils/utils.ts (hunk 3 above)
Comment (asanchezr) [reply frontend/src/utils/utils.ts:1]: tech-debt yes

File: frontend/src/components/common/form/Select.tsx (hunk 2 above)
Comment (asanchezr) [reply frontend/src/components/common/form/Select.tsx:1]: The select supports passing it an `options` array so yes that's a dynamic list. See the usage in `MapFilterBar.tsx`

```tsx
<Select
    field="classificationId"
    placeholder="View by Classification"
    options={classifications}
```

where the `classifications` array comes from a previous call to the API.
Hope that answers your question...

**Pull request merged**